### ISSUE TITLE
Can't publish since last update

### ISSUE THREAD

**Issue by Germain-Italic** [chrishasz/spgo#106]
Hi, me and my team are facing a weird issue since last week (say Tuesday or Friday).
I can see that there were new commits 5 days ago so I'm wondering if it's related.

Basically we can't publish files anymore.
We're facing the issue on different sites/tenants/accounts, so certain it's not an account problem.

We can save : 

```
[09/12/2019 à 17:04:38] SPGo enabled.
[09/12/2019 à 17:04:43] Getting file information for:  c:\...\.css
[09/12/2019 à 17:04:45] Check out Type: 2
[09/12/2019 à 17:04:45] File is not checked out.
[09/12/2019 à 17:05:12] Saving file:  c:\...\.css
[09/12/2019 à 17:05:13] File saved successfully
```
We have this "File is not checked out." message that did not occur before.
Of course we try to checkout the file: nothing happens.

We try to publish the file: nothing happens.

In short, we're stuck.

Is there any way to turn back to a previous version?

**Comment by Germain-Italic** [comment]
Just to confirm, current version is:
1.5.0 - 2019-12-03 => not working

Previous version was:
1.4.5 - 2019-10-22 => working

**Comment by Germain-Italic** [comment]
I've revert to 1.4.5 by opening the Extensions panel then right-click on SPGo > install other version.

Can confirm the publishing is working again: 

```
Publishing Major file version:  c:\...\.css
Saving file:  c:\...\.css
File publish complete.
```

Feel free to ask anything if it helps for debugging.

**Comment by chrishasz** [comment]
Hello @Germain-Italic 

Couple of things to help debug here:

I assume this is SharePoint online using Digest authentication?

1) Can you post your `spgo.json` config file? (feel free to mask the tenant name e.g. https://xxx.sharepoint.com)
2) Are you using Windows, or OSX/Linux?
3) How are you publishing the file? Are you using the `Publish Minor` or `Publish Major` commands, or are you using the automated publish file on save feature? If you are using the commands, are you using the treeview, hotkey, or VSCode command window?
4) do you see any errors in the console log when you enable VSCode developer window? You can enable this through the menu on windows OS by going `Help->Toggle Developer Tools"

Regarding the output messages, I enhanced logging in 1.5.0 to provide more data to help debug as well as adding timestamps for the messages to better understand which events are grouped. The message you shared come from two different groups.
The first group shows SPGo successfully loading, and then you opened a `.css` file. SPGo queries the server for information about all files that are opened for edit. In this case, you can see that the file is not checked out. A Check out Type of "2" is the enum for "Not checked out" per the [documentation](https://docs.microsoft.com/en-us/previous-versions/office/sharepoint-csom/ee538918(v%3Doffice.15))
```
[09/12/2019 à 17:04:38] SPGo enabled.
[09/12/2019 à 17:04:43] Getting file information for:  c:\...\.css
[09/12/2019 à 17:04:45] Check out Type: 2
[09/12/2019 à 17:04:45] File is not checked out.
```

The second group shows that a file is saved successfully to the server, but it not published. Did you use the publish command here? If not, it looks like there may be a bug in how Publishing works when automatic publishing is enabled on save.
```
[09/12/2019 à 17:05:12] Saving file:  c:\...\.css
[09/12/2019 à 17:05:13] File saved successfully
```

I'll look into auto-publishing, if you can help answer the questions above.

**Comment by Germain-Italic** [comment]
> I assume this is SharePoint online using Digest authentication?

Correct

> Can you post your spgo.json config file? 

```
{
    "sourceDirectory": "src",
    "sharePointSiteUrl": "https://xxxxxx.sharepoint.com/sites/xxxxxxxxxx",
    "publishingScope": "SaveOnly",
    "authenticationType": "Digest",
    "storeCredentials": true,
    "remoteFolders": [
        "/_catalogs/masterpage/scripts/",
        "/_catalogs/masterpage/styles/",
        "/_catalogs/masterpage/images/**/*",
        "/_catalogs/masterpage/Prévoir/**/*.css",
        "/_catalogs/masterpage/Prévoir/**/*.js",
        "/_catalogs/masterpage/Prévoir/**/*.html",
        "/_catalogs/masterpage/Prévoir/**/*.aspx",
        "/_catalogs/masterpage/Prévoir/**/*.master"
    ]
}
```

> Are you using Windows, or OSX/Linux?

3 devs using Windows 10 latest stable update. Should this help:

![](https://files.italic.fr/Code_qu4tEnFiib.png)

> How are you publishing the file? 

`Publish major` using hotkey and treeview + right-click depending on the user.

> do you see any errors in the console log

Debug console was already opened and checked, it stays blank.

> The second group shows that a file is saved successfully to the server, but it not published. Did you use the publish command here? 

Nope it's the default action from SPGo.json, we always save automatically, publish major manually (we don't use publish minor, however we've tried it for debug purpose and it didn't work either, ie. not output message in SPGo output or debug console).

Good luck!

**Comment by PopWarner** [comment]
I can confirm that v1.50 is failing also when you even try to "Configure workspace".

I was able to rollback to v1.4.5 and it works successfully.

**Comment by chrishasz** [comment]
@Germain-Italic @PopWarner - I'm really having a hard time reproducing this issue. 

If I understand what you are seeing correctly:
- SPGo activates successfully (`SPGo Enabled.` message in the SPGo output).
- The commands are available
- When the command is run from either the context menu, or the command runner (<ctrl>+<shift>+p), there are messages output to the SPGo output logs, but no errors.
- When you open VSCode developer mode, there are no errors in the JS console

Is this what both of you are seeing? Are there any other details that I'm missing which can help me reproduce?

**Comment by Germain-Italic** [comment]
Hi @chrishasz, that's correct except:

> When the command is run from either the context menu, or the command runner (++p), there are messages output to the SPGo output logs, but no errors.

Nope.  Saving files works and show the log but publishing does not do / output anything.
Also "Populating local workspace" spins continuously sometimes in the bottom left blue bar after the first save/populate.

**Comment by PopWarner** [comment]
Same for me as @Germain-Italic. On a brand new machine, Windows, installed the most recent version of VSCode and SPGo and VSCode open to a folder.

I then am able to F1, type SPG, then select "Configure Workspace" and nothing happens. No output, almost as if I didn't even select it.

I installed the previous version and perform the same steps and it worked as expected.

**Comment by chrishasz** [comment]
@PopWarner - I am able to reproduce the issue you are seeing with a clean setup on a new folder. I was also able to get the “Configure Workspace” command to work if I ran it twice in a row. Can you confirm if this works for you? I’ll fix this in 1.5.1.

@Germain-Italic - can you show me the vscode output from the dev console when you run the “Publish Major” command? A screenshot is fine.

Thanks all! I’m hoping to get this resolved soon with your help.

**Comment by Germain-Italic** [comment]
@chrishasz the Debug Console always stays empty.

When I run from F1 > Publish major: nothing happens.

When I run from file explorer > right click > publish major: seems to work normally:

```
[12/12/2019 à 21:30:37] SPGo enabled.
[12/12/2019 à 21:32:21] Publishing Major file version:  c:\xxx.css
[12/12/2019 à 21:32:32] Saving file:  c:\xxx.css
[12/12/2019 à 21:32:36] File publish complete.
```

I have to admin that I did not try to use the Publish feature using contextual menu before tonight!

**Comment by PopWarner** [comment]
@chrishasz, I just tested what works for you, in running "Configure Workspace" twice in a row on a clean folder setup and it does not work. :( This is with SPGO 1.5 and the follwoing VSCode version:
![image](https://user-images.githubusercontent.com/10676147/70748587-260ea480-1cdf-11ea-8cb2-9e3fed68ea6e.png)

**Comment by chrishasz** [comment]
@Germain-Italic @PopWarner - I have now successfully reproduced both issues locally. Thank you both for the help. I will get these issues resolved for 1.5.1.

@Germain-Italic are you able to use the option of publishing from the context menu as a workaround until 1.5.1 is released?

**Comment by Germain-Italic** [comment]
Hi we rolled back to 1.4.x

**Comment by chrishasz** [comment]
@Germain-Italic, @PopWarner - I released SPGo 1.5.1 to the VSCode Marketplace last night which will resolve this issue. 

Please check it out and let me know if the bugs are resolved for you and I will close this issue.

**Comment by PopWarner** [comment]
Yay!! I can confirm success in the "Configure Workspace" functionality. I will do a complete test asap and report back. Thanks, @chrishasz for the swift corrections! 👍

**Comment by Germain-Italic** [comment]
Hello, "Publish a major version of the current file" seems to be working! Congratulations

**Comment by pfrowe** [comment]
"Publish a minor version of the current file" worked for me in 1.5.1 where it did not in 1.5.0. Thanks!

**Comment by Germain-Italic** [comment]
@chrishasz just to let you know that Palette > Get server version is not working, with no log/error message. Right click > Get server version works however.

Using 1.5.1.

**Comment by chrishasz** [comment]
Thanks for letting me know @Germain-Italic. I'll get this turned around in the next few days.

**Comment by chrishasz** [comment]
OK, I got motivated tonight and I fixed this bug in version1.5.2, which is available now on the VSCode marketplace.

I also added better documentation for the [multiple workspaces](https://www.chrishasz.com/spgo/advanced/multiple-workspaces) and [SubSites](https://www.chrishasz.com/spgo/advanced/working-with-subsites) features.

**Comment by Germain-Italic** [comment]
Confirmed as working, congratulations @chrishasz 

```
[20/12/2019 à 20:13:51] Getting server version for:  020-user.js
[20/12/2019 à 20:13:51] Successfully downloaded 1 files to: c:\Users\...
```

**Comment by chrishasz** [comment]
Looks like everything is stable. Closing.
